Hazel/src/Platform/Windows/WindowsWindow.cpp

```diff
@@ -55,6 +58,9 @@ namespace Hazel {
 
 		glfwSetWindowUserPointer(m_Window, &m_Data);
 		SetVSync(true);
+		
+		glfwGetWindowPos(m_Window, &(m_Data.XPos), &(m_Data.YPos));
+		SetWindowMode(props.Mode);
 
 		// Set GLFW callbacks
```

Comment:
We now have this nice function, but we're not utilizing it.

Below the `glfwSetWindowSizeCallback` we should add [`glfwSetWindowMaximizeCallback`](https://www.glfw.org/docs/latest/group__window.html#gaf8548ef61cb09150e09a6d33ddfa705e) with a lamda callback function with [this signature](https://www.glfw.org/docs/latest/group__window.html#ga7269a3d1cb100c0081f95fc09afa4949) which will switch between windowed and fullscreen mode, something like:

```cpp
glfwSetWindowMaximizeCallback(m_Window, [](GLFWwindow* window, int isMaximised)
{
	switch (isMaximised)
	{
		case GLFW_TRUE:
		{
			// update data to fullscreen mode
			return;
		}
		case GLFW_FALSE:
		{
			// update data to windowed mode
			return;
		}
	}
	HZ_CORE_ERROR("GLFW invalid window maximize callback);
}
```
Note: here we can't call `SetWindowMode` as that would set the glfw window again, here we just want to update our m_Data...

Reply:
Fixed in https://github.com/TheCherno/Hazel/pull/108/commits/26be06458300995e9532267d3c76cce9028fb29f